### PR TITLE
Fix failing CI on MDAnalysis benchmark

### PR DESCRIPTION
**Pull request by bdice**
This PR fixes failing CI by updating to the syntax of MDAnalysis 2.0. The notebook was using a deprecated method `as_Universe`, which was removed in MDAnalysis 2.0.

~I also updated the notebook contents a bit. When this notebook was originally written, MDAnalysis (pre-1.0) had a slower implementation of the RDF than freud. MDAnalysis now ships with OpenMP-parallelized C code for distance computations that is moderately faster than freud, especially for small systems. It's helpful to present the benchmark regardless of the "winner" so that users can see how the syntax compares and choose the best tool for their needs. Congrats to the MDAnalysis team on their upgrades, of course!~

edit: @vyasr noticed that this performance regression was due to freud missing optimization flags. This was corrected in freud 2.7.0. See: https://github.com/glotzerlab/freud/pull/843

### CODE REVIEW

**Comment by vyasr** [comment]
@bdice in light of glotzerlab/freud#843 you'll probably want to update the PR description and rerun. You may also want to add back the "speedup" cell that you had previously deleted.

**Comment by bdice** [comment]
I plan to merge this when CI passes, since our CI is currently broken.

File: examples/Benchmarking RDF against MDAnalysis.ipynb [hunk 1]
```diff
@@ -5,7 +5,7 @@
    "metadata": {},
    "source": [
     "# Benchmarking RDF against MDAnalysis\n",
-    "The algorithms in freud are highly efficient and rely on parallelized C++ code. Below, we show a benchmark of `freud.density.RDF` against `MDAnalysis.analysis.rdf`. This benchmark was run on an Intel(R) Core(TM) i3-8100B CPU @ 3.60GHz."
+    "The algorithms in freud are highly efficient and rely on parallelized C++ code. Below, we show a benchmark of  `freud.density.RDF` and `MDAnalysis.analysis.rdf.InterRDF`. The algorithms in both packages are efficient and rely on parallelized C or C++ code. This benchmark was run on an Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz."
```
Comment (bdice) [inline examples/Benchmarking RDF against MDAnalysis.ipynb:8]:
```suggestion
    "The algorithms in freud are highly efficient and rely on parallelized C++ code. Below, we show a benchmark of `freud.density.RDF` and `MDAnalysis.analysis.rdf.InterRDF`. This benchmark was run on an Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz."
```

**Comment by bdice** [comment]
The remaining CI failure is not related to the fix in this PR. Merging.

**Pull request merged**